lib/cn/kvset.c

```diff
@@ -2557,21 +2549,17 @@ kvset_iter_enable_mblock_read(struct kvset_iterator *iter)
     if (ra_size < HSE_KVS_VALUE_LEN_MAX) {
         if (iter->ks->ks_node_level == 0)
             ra_size = HSE_KVS_VALUE_LEN_MAX;
-        ra_size = min_t(uint, ra_size, HSE_KVS_VALUE_LEN_MAX);
+        ra_size = min_t(uint64_t, ra_size, HSE_KVS_VALUE_LEN_MAX);
     }
 
     /* Limit buffered reads to values lesser than cn_compact_vblk_ra. The
      * upper level spill/compaction routines make direct reads for the sizes
      * matching or exceeding cn_compact_vblk_ra.
      */
-    vr_buf_sz = max_t(uint, PAGE_SIZE, ra_size);
-
-    vb_max_sz = iter->ks->ks_rp->vblock_size;
-    if (vb_max_sz > VLB_ALLOCSZ_MAX / 2)
-        vb_max_sz = VLB_ALLOCSZ_MAX / 2;
+    vr_buf_sz = max_t(uint64_t, PAGE_SIZE, ra_size);
```

Comment:
Since ra_size should never be less than 32K it seems this line could just be:   vr_buf_sz = ra_size;